nagios/overall.go

```diff
@@ -44,27 +44,25 @@ func (o *Overall) GetStatus() int {
 }
 
 func (o *Overall) GetSummary() string {
+	if o.Summary != "" {
+		return o.Summary
+	}
+	if o.Criticals > 0 {
+		o.Summary += fmt.Sprintf("critical=%d ", o.Criticals)
+	}
+	if o.Unknowns > 0 {
+		o.Summary += fmt.Sprintf("unknown=%d ", o.Unknowns)
+	}
+	if o.Warnings > 0 {
+		o.Summary += fmt.Sprintf("warning=%d ", o.Warnings)
+	}
+	if o.OKs > 0 {
+		o.Summary += fmt.Sprintf("ok=%d ", o.OKs)
+	}
 	if o.Summary == "" {
-		if o.Criticals > 0 {
-			o.Summary += fmt.Sprintf("critical=%d ", o.Criticals)
-		}
-		if o.Unknowns > 0 {
-			o.Summary += fmt.Sprintf("unknown=%d ", o.Unknowns)
-		}
-		if o.Warnings > 0 {
-			o.Summary += fmt.Sprintf("warning=%d ", o.Warnings)
-		}
-		if o.OKs > 0 {
-			o.Summary += fmt.Sprintf("ok=%d ", o.OKs)
-		}
-		if o.Summary == "" {
-			o.Summary = "No status information"
-		} else {
-			o.Summary = "states: " + strings.TrimSpace(o.Summary)
-		}
+		return "No status information"
 	}
-
-	return o.Summary
+	return "states: " + strings.TrimSpace(o.Summary)
```

Comment:
We should store it in o.Summary and return this.

```suggestion
	o.Summary = "states: " + strings.TrimSpace(o.Summary)
	return o.Summary
```

Reply:
Argh, sorry, of course! Thanks for catching this :-/